zcash_client_sqlite/src/lib.rs

```diff
@@ -401,18 +403,21 @@ impl<P: consensus::Parameters> WalletWrite for WalletDb<rusqlite::Connection, P>
         blocks: Vec<ScannedBlock<sapling::Nullifier>>,
     ) -> Result<Vec<Self::NoteRef>, Self::Error> {
         self.transactionally(|wdb| {
-            let start_position = blocks.first().map(|block| {
-                Position::from(
-                    u64::from(block.metadata().sapling_tree_size())
-                        - u64::try_from(block.sapling_commitments().len()).unwrap(),
+            let start_positions = blocks.first().map(|block| {
+                (
+                    block.height(),
+                    Position::from(
+                        u64::from(block.metadata().sapling_tree_size())
+                            - u64::try_from(block.sapling_commitments().len()).unwrap(),
+                    ),
                 )
             });
             let mut wallet_note_ids = vec![];
             let mut sapling_commitments = vec![];
-            let mut end_height = None;
-
+            let mut last_scanned_height = None;
+            let mut note_positions = vec![];
             for block in blocks.into_iter() {
-                if end_height.iter().any(|prev| block.height() != *prev + 1) {
+                if last_scanned_height.iter().any(|prev| block.height() != *prev + 1) {
```

Comment:
In Rust 1.70.0 there is [`Option::is_some_and`](https://doc.rust-lang.org/std/option/enum.Option.html#method.is_some_and):
```rust
                if last_scanned_height.is_some_and(|prev| block.height() != *prev + 1) {
```
but the MSRV is currently 1.65.